app/src/main/java/com/fastlink/zekrapp/presentation/ZekirScreen.kt

```diff
@@ -0,0 +1,198 @@
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.FabPosition
+import androidx.compose.material.FloatingActionButton
+import androidx.compose.material.FloatingActionButtonDefaults
+import androidx.compose.material.Scaffold
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.platform.LocalClipboardManager
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.fastlink.zekrapp.LocalToast
+import com.fastlink.zekrapp.LocalViewModel
+import com.fastlink.zekrapp.presentation.utils.DashedDevider
+import com.fastlink.zekrapp.presentation.utils.appBars.ZekirScreenAppBar
+import com.fastlink.zekrapp.presentation.utils.archBorder
+import com.fastlink.zekrapp.presentation.utils.bottomAppBar.ZekirScreenBottomAppBar
+import com.fastlink.zekrapp.presentation.utils.dashedBorder
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+
+@SuppressLint(
+    "UnrememberedMutableInteractionSource", "CoroutineCreationDuringComposition",
+    "ShowToast"
+)
+@Composable
+fun ZekirScreen(categoryId: Int) {
+    val toast = LocalToast.current
+    val clipboardManager = LocalClipboardManager.current
+    val viewModel = LocalViewModel.current
+    val scope = rememberCoroutineScope()
+    val category = viewModel.getCategoryById(categoryId)
+    val zekirs = viewModel.getZekirByCategoryId(category.id)
+    val zekirNumber = remember {
+        mutableIntStateOf(0)
+    }
+    val zekirCounter = remember {
+        mutableIntStateOf(0)
+    }
+    LaunchedEffect(true) {
+        toast(" الذكر ${zekirNumber.intValue + 1} من ${zekirs.size}")
+    }
+    val animatedBorder by animateFloatAsState(
+        targetValue =
+        (zekirCounter.intValue.toFloat() / zekirs[zekirNumber.intValue].counterNumber.toFloat()) * 360f,
+        label = "",
+        animationSpec = tween(1000)
+    )
+
+    fun onClicked() {
```

Comment:
Here if you read this function name you don't know what it's doing, clicked on what? Also you should make the composable for showing the ui, handling the logic and state management should be in the view model